cartographer_rviz/cartographer_rviz/drawable_trajectory.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_RVIZ_SRC_DRAWABLE_TRAJECTORY_H_
+#define CARTOGRAPHER_RVIZ_SRC_DRAWABLE_TRAJECTORY_H_
+
+#include <future>
+
+#include "rviz/display_context.h"
+#include "rviz/frame_manager.h"
+#include "rviz/properties/bool_property.h"
+
+#include "cartographer/mapping/id.h"
+#include "cartographer_rviz/drawable_submap.h"
+
+namespace cartographer_rviz {
+class DrawableTrajectory : public QObject {
```

Comment:
Misleading name, since it contains Submaps and does not really draw the trajectory. PerTrajectorySubmapDisplay?